Updated header to match new design

chrisrhymes opened the pull request:
Resolves part of the issues for: https://github.com/FutureNorthants/northants-website/issues/1102

There is a lot of changes in this pull request:
- Header redesign
- HomeHero allowing to hide the search and have the promoted links as tabs at the top
- PromotedServicesTabs to show at the bottom of the HomeHero
- All of it combined in the Homepage example

## Testing 
- Checkout this branch and run `npm install` then `npm run dev`
- Test the styles and interactivity for the Header, PromotedLinks and PromotedServicesTabs
- PromotedServicesTabs should have the relevant aria attributes for the [ARIA tab role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role).
- Test the HomeHero updates with the promoted services tabs and showSearch set to false
- Test it all combined in the Homepage example page. 
- Manually run tests with `npm run test`

chrisrhymes commented on src/library/structure/HomeHero/HomeHero.storydata.ts:
Replaced with picsum image links as the placehold.it images didn't always load.

chrisrhymes commented on src/library/structure/HomeHero/HomeHero.tsx:
```diff
@@ -21,21 +24,24 @@ const HomeHero: React.FunctionComponent<HomeHeroProps> = ({
-      <Styles.Wrapper>
-        <Styles.Container className={random !== 999 ? 'loaded' : 'loading'}>
+    <div data-testid="HomeHero">
+      <Styles.Wrapper as={showSearch ? 'header' : 'div'}>
```
The `as` changes the html element used based on the prop passed in. When showSearch is false we need to show the header component, so we want this to be a div and not have two HTML header elements on the page.

chrisrhymes commented on src/library/structure/PromotedServicesTabs/PromotedServicesTabs.stories.tsx:
```diff
@@ -0,0 +1,36 @@
+    <SBPadding>
+      <MaxWidthContainer>
+        <PageMain>
+          <PromotedServicesTabs {...args} activeTab={activeTab} setActiveTab={setActiveTab} />
```
The tabs and the tab content are split into two components. This is so they can be inserted into different sections of the HomeHero component for the design layout to work.

chrisrhymes commented on src/library/structure/PromotedServicesTabs/PromotedServicesTabContent.tsx:
```diff
@@ -0,0 +1,56 @@
+    <Styles.TabContentContainer data-testid="PromotedServicesTabContent">
+      {tabs.map((tab, contentIndex) => (
+        <Styles.TabContentInner key={contentIndex}>
+          <Styles.TabMobile
```
On mobiles the PromotedServicesTabs component is hidden and these TabMobile are shown instead. They act more like an accordion in this situation than tabs where you click to expand the TabContent and show the links.

chrisrhymes commented on src/library/components/PromotedLinks/PromotedLinks.tsx:
The behaviour of this component is modified based on the `hasTabs` prop. If it's false of undefined it will act the same as it did before. If `hasTabs` is true then it will display smaller tabs that are left aligned.

chrisrhymes commented on src/library/components/logos/NorthBlackLogo/logo.tsx:
New black logos used for the memorial theme header styles.

chrisrhymes commented on src/library/pages/CouncilTaxPageExample/CouncilTaxPageExample.tsx:
```diff
@@ -3,156 +3,158 @@ import * as PageStructures from './../../structure/PageStructures';
-      <PageStructures.Breadcrumbs 
-        breadcrumbsArray={[
+    <>
+      <PageStructures.Header hasDirectoryLink hasNewsLink accessibilityLink="/" />
```
Change is adding the props to the header. The rest is prettier formatting.

chrisrhymes commented on src/library/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -1,63 +1,65 @@
-  <PageStructures.Header />
-  <PageStructures.MaxWidthContainer>
+  <>
+    <PageStructures.Header hasDirectoryLink hasNewsLink accessibilityLink="/" />
```
Change is adding the props to the header. The rest is prettier formatting.

chrisrhymes commented on src/library/structure/Header/Header.styles.js:
```diff
@@ -169,14 +201,83 @@ export const Link = styled.a`
 `;
 
 export const SearchWrapper = styled.div`
-  grid-area: headersearch;
```
Replaced CSS grid styles with flexbox.

chrisrhymes commented:
> Discovered a bug with the search bar suggestions not displaying due to z-index overlap with the navbar below. 
This is fixed in the Google Translate pull request: https://github.com/FutureNorthants/northants-design-system/pull/536#discussion_r1848028349

mstanford-wnc commented:
When the **'Promoted Services'** item is viewed on a mobile phone (e.g. an iPhone SE 3rd Gen) there are no grey lines on the sides, whereas there are light grey lines on the top and bottom

![image](https://github.com/user-attachments/assets/8dd04467-58b4-4339-9f91-6cce42f6a0f6)

and when the first item is selected by default (as below), there is no light grey line at the top either:

![image](https://github.com/user-attachments/assets/32ba2f48-b8ca-457b-ad5f-de5769ea9518)

and when the section name is highlighted in orange (as below), the transition between the orange and grey sections doesn't 
appear to be as seamless as would perhaps be desired:   (Though this might just be a quirk of our 'Storybook' website)

![image](https://github.com/user-attachments/assets/6c708cb0-66c5-4d94-a12a-8043b322872e)

mstanford-wnc commented:
**Promoted Links** do not seem to have top and side lines that are a dark enough shade of grey or blue (or another colour of choice), whether viewed on a smartphone or a regular browser.

![image](https://github.com/user-attachments/assets/ab54a999-f40e-4842-a6ba-9dc4c139fda7)
![image](https://github.com/user-attachments/assets/4a1fd569-2ef6-4ac7-807e-955e863b5e6b)

chrisrhymes commented:
> When the **'Promoted Services'** item is viewed on a mobile phone (e.g. an iPhone SE 3rd Gen) there are no grey lines on the sides, whereas there are light grey lines on the top and bottom
> 

The light blue background for the active tab and tab content matches the styles for the desktop version. These tabs don't have grey borders. My personal opinion is that they are not needed as the light blue separates from the white outer background. Sometimes less is more and adding unnecessary borders can make things look too busy.  

> and when the first item is selected by default (as below), there is no light grey line at the top either:
> 

I have added a border-top to the first item so it looks the same as the other tabs on mobile.

> and when the section name is highlighted in orange (as below), the transition between the orange and grey sections doesn't appear to be as seamless as would perhaps be desired: (Though this might just be a quirk of our 'Storybook' website)
> 

I have moved the z-index higher so the focus state now shows the black border all the way around the focused tab.

chrisrhymes commented:
> **Promoted Links** do not seem to have top and side lines that are a dark enough shade of grey or blue (or another colour of choice), whether viewed on a smartphone or a regular browser.
> 

These are used for a slightly different purpose than the other tabs and links so they look a bit different. In the new design they hang down from the top of the header bar like so. This is why they don't have a top border. 
![Screenshot 2024-11-20 at 10 08 30](https://github.com/user-attachments/assets/e179e9e3-9492-4b47-9307-9c8a9d9555e9)

chrisrhymes commented on src/library/structure/PromotedServicesTabs/PromotedServicesTabs.styles.js:
```diff
@@ -0,0 +1,148 @@
+  flex-wrap: nowrap;
+  justify-content: space-between;
+  align-items: center;
+  text-align: left;
```
Ensure text is left align if it wraps onto a second line.

chrisrhymes commented on src/library/structure/PromotedServicesTabs/PromotedServicesTabs.styles.js:
```diff
@@ -0,0 +1,148 @@
+
+export const ServiceLink = styled.a`
+  display: block;
+  height: 100%;
```
Make link grow in height to match the largest link in the row.

mstanford-wnc reviewed (approved):
Looks okay to me
Approving this
